### PR TITLE
Add codespell workflow and fix typos

### PR DESCRIPTION
**Pull request by dnkolegov**
This PR:

-  fixes multiple typos in the documentation, comments and code
- adds `codespell` workflow

The main goal is to make the code and documentation more clean.

Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs)
- [x]  Tests for the changes have been added / updated
- [x]  Documentation comments have been added / updated.

### CODE REVIEW

File: .codespell/.codespellrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[codespell]
+skip = .git,node_modules,artifacts-zk,yarn.lock,contracts-preprocessed,Cargo.lock
+count =
```
Comment (StanislavBreadless) [inline .codespell/.codespellrc:3]: I am not knowledgable about codespell, but it looks like a field that should be filled

File: l1-contracts/test/foundry/unit/concrete/Bridges/L1SharedBridge/L1SharedBridgeFails.t.sol [hunk 2]
```diff
@@ -92,7 +92,7 @@ contract L1SharedBridgeFailTest is Test {
     uint16 l2TxNumberInBatch;
     bytes32[] merkleProof;
 
-    // storing depoistHappend[chainId][l2TxHash] = txDataHash. DepositHappened is 3rd so 3 -1 + dependency storage slots
+    // storing depositHappend[chainId][l2TxHash] = txDataHash. DepositHappened is 3rd so 3 -1 + dependency storage slots
```
Comment (StanislavBreadless) [inline l1-contracts/test/foundry/unit/concrete/Bridges/L1SharedBridge/L1SharedBridgeFails.t.sol:95]: I would rather rename it to `depositHappened`, not sure why it wasn't detected

**Review by StanislavBreadless — COMMENTED** [review]
One skipped issue and after that we are good to go

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/RevertBatches.t.sol [hunk 3]
```diff
@@ -36,7 +36,7 @@ contract revertBatchesTest is StateTransitionManagerTest {
         gettersFacet = GettersFacet(address(newChainAddress));
         adminFacet = AdminFacet(address(newChainAddress));
 
-        // Initital setup for loge & commits
+        // Initial setup for loge & commits
```
Comment (StanislavBreadless) [inline l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/RevertBatches.t.sol:39]: "loge" should be fixed also

File: l1-contracts/test/foundry/unit/concrete/Bridges/L1SharedBridge/L1SharedBridgeFails.t.sol (hunk 2 above)
Comment (dnkolegov) [reply l1-contracts/test/foundry/unit/concrete/Bridges/L1SharedBridge/L1SharedBridgeFails.t.sol:95]: Fixed

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/RevertBatches.t.sol (hunk 3 above)
Comment (dnkolegov) [reply l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/RevertBatches.t.sol:39]: Fixed

File: .codespell/.codespellrc (hunk 1 above)
Comment (dnkolegov) [reply .codespell/.codespellrc]: Fixed

**Review by StanislavBreadless — APPROVED** [review]

**Pull request merged**